src/outlineView.ts

```diff
@@ -29,23 +46,63 @@ export class OutlineView {
     this.outlineContent.classList.add("outline-content")
   }
 
-  destroy() {
+  public destroy() {
     this.element.remove()
   }
 
-  getElement() {
+  public getElement() {
     return this.element
   }
 
-  getTitle() {
+  public getTitle() {
     return "Outline"
   }
 
-  getIconName() {
+  public getIconName() {
     return "list-unordered"
   }
 
-  setOutline(outlineTree: OutlineTree[], editor: TextEditor, isLarge: boolean) {
+  public setOutline(editor: TextEditor | undefined) {
+    return new Promise<void>((resolve) => {
```

Comment:
Another simpler approach is to not care about when this is resolved, and use a `sleep` function in the tests. I actually started with that. But jasmine wasn't awaiting the `sleep` function either.

```js
function sleep(ms = 1000) {
	return new Promise((resolve) => {
		setTimeout(resolve, ms)
	})
}
```

Reply:
One other thing is to make a timer using a while loop. I haven't tried that though

```js
  function sleep(delay) {
    const start = new Date().getTime();
    while (new Date().getTime() < start + delay) {
    	// do nothing
    };
  }

```